src/langchain_google_alloydb_pg/pg_to_alloy_migrator.py

```diff
@@ -0,0 +1,529 @@
+            return result_fetch
+        except:
+            raise ValueError(f"Collection: {collection_name} does not exist.")
+
+    async def _ainsert_single_batch(
+        self,
+        destination_table: str,
+        metadata_column_names: Optional[List[str]],
+        data: Sequence[RowMapping],
+        use_json_metadata: Optional[bool] = False,
+    ) -> None:
+        """
+        Create a batch insert SQL query to insert multiple rows at once.
+        Looks like
+        INSERT INTO MyTable ( Column1, Column2 )
+        VALUES ( Value1, Value2 ), ( Value1, Value2 ), ...
+
+        Args:
+            destination_table (str): The name of the table to insert the data in.
+            metadata_column_names (str): The metadata columns to be created to keep the data in a row-column format.
+                Optional.
+            data (Sequence[RowMapping]): All the data (to be inserted) belonging to a pgvector collection.
+            use_json_metadata (bool): An option to keep the PGVector metadata as json in the AlloyDB table.
+                Default: False. Optional.
+        """
+        row_number = 0
+        params = {}
+
+        if use_json_metadata:
+            insert_query = f"INSERT INTO {destination_table} (langchain_id, content, embedding, langchain_metadata) VALUES"
+
+            # Create value clause for the SQL query
+            values_clause = ", ".join(
+                [
+                    f"(:langchain_id_{num}, :content_{num}, :embedding_{num}, :langchain_metadata_{num})"
+                    for num in range(len(data))
+                ]
+            )
+            insert_query += values_clause
+
+            # Add parameters
+            for row_number in range(len(data)):
+                row = data[row_number]
+                params[f"langchain_id_{row_number}"] = row.id
+                params[f"content_{row_number}"] = row.document
+                params[f"embedding_{row_number}"] = row.embedding
+                params[f"langchain_metadata_{row_number}"] = json.dumps(row.cmetadata)
+
+                row_number += 1
+
+        elif metadata_column_names:
+            insert_query = (
+                f"INSERT INTO {destination_table} (langchain_id, content, embedding"
+            )
+            for column in metadata_column_names:
+                insert_query += ", " + column
+            insert_query += ") VALUES"
+
+            # Create value clause for the SQL query
+            values_clause = ", ".join(
+                [
+                    f"(:langchain_id_{num}, :content_{num}, :embedding_{num}, "
+                    + ", ".join(
+                        [f":{column}_{num}" for column in metadata_column_names]
+                    )
+                    + ")"
+                    for num in range(len(data))
+                ]
+            )
+            insert_query += values_clause
+
+            # Add parameters
+            for row in data:
```

Comment:
Can we clean up this for loop similar to above?

Reply:
Done